From 6dbe8b8e9a2fc7d5305beaa0c91538f60a186491 Mon Sep 17 00:00:00 2001
From: "Yuanbin.Chen" <cybing4@gmail.com>
Date: Wed, 20 Sep 2017 22:48:58 +0800
Subject: [PATCH 2/4] Add trove database mysql monitor

Feature-ES #11008
http://192.168.15.2/issues/11008

Signed-off-by: Yuanbin.Chen <cybing4@gmail.com>
(cherry picked from commit 9bfabb2a8dca1e756b693e9fd170555be91fd524)
---
 ceilometer/database/mysql.py | 208 +++++++++++++++++++++++++++++++++++++++++++
 setup.cfg                    |  21 +++++
 2 files changed, 229 insertions(+)
 create mode 100644 ceilometer/database/mysql.py

diff --git a/ceilometer/database/mysql.py b/ceilometer/database/mysql.py
new file mode 100644
index 0000000..cf896fc
--- /dev/null
+++ b/ceilometer/database/mysql.py
@@ -0,0 +1,208 @@
+#
+# Copyright 2017 Eayun, Inc.
+#
+# Author: Julien Danjou <julien@danjou.info>
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+from ceilometer import sample
+from notifications import DataBaseNotificationBase
+
+
+class MysqlTotalConnectPollster(DataBaseNotificationBase):
+    """Listen for Trove notifications.
+       Listen in order to mediate with the metering framework.
+    """
+    resource_name = 'mysql.total.connections'
+    unit = 'connect'
+    sample_type = sample.TYPE_CUMULATIVE
+
+
+class MysqlThreadsConnectPollster(DataBaseNotificationBase):
+    """Listen for Trove notifications.
+       Listen in order to mediate with the metering framework.
+    """
+    resource_name = 'mysql.threads.connections'
+    unit = 'connect'
+    sample_type = sample.TYPE_DELTA
+
+
+class MysqlQPSPollster(DataBaseNotificationBase):
+    """Listen for Trove notifications.
+       Listen in order to mediate with the metering framework.
+    """
+    resource_name = 'mysql.qps'
+    unit = 'request'
+    sample_type = sample.TYPE_CUMULATIVE
+
+
+class MysqlQPSRatePollster(DataBaseNotificationBase):
+    """Listen for Trove notifications.
+       Listen in order to mediate with the metering framework.
+    """
+    resource_name = 'mysql.qps.rate'
+    unit = 'request/s'
+    sample_type = sample.TYPE_DELTA
+
+
+class MysqlTPSPollster(DataBaseNotificationBase):
+    """Listen for Trove notifications.
+       Listen in order to mediate with the metering framework.
+    """
+    resource_name = 'mysql.tps'
+    unit = 'request'
+    sample_type = sample.TYPE_CUMULATIVE
+
+
+class MysqlTPSRatePollster(DataBaseNotificationBase):
+    """Listen for Trove notifications.
+       Listen in order to mediate with the metering framework.
+    """
+    resource_name = 'mysql.tps.rate'
+    unit = 'request/s'
+    sample_type = sample.TYPE_DELTA
+
+
+class MysqlSlowQuerysPollster(DataBaseNotificationBase):
+    """Listen for Trove notifications.
+       Listen in order to mediate with the metering framework.
+    """
+    resource_name = 'mysql.slow.querys'
+    unit = 'request'
+    sample_type = sample.TYPE_CUMULATIVE
+
+
+class MysqlSyncSlaveDelayPollster(DataBaseNotificationBase):
+    """Listen for Trove notifications.
+       Listen in order to mediate with the metering framework.
+    """
+    resource_name = 'mysql.sync.delay'
+    unit = 's'
+    sample_type = sample.TYPE_DELTA
+
+
+class MysqlScanFullTablePollster(DataBaseNotificationBase):
+    """Listen for Trove notifications.
+       Listen in order to mediate with the metering framework.
+    """
+    resource_name = 'mysql.scan.full.table'
+    unit = 'request'
+    sample_type = sample.TYPE_CUMULATIVE
+
+
+class MysqlBufferPoolDirtyRatioPollster(DataBaseNotificationBase):
+    """Listen for Trove notifications.
+       Listen in order to mediate with the metering framework.
+    """
+    resource_name = 'mysql.buffer.pool.dirty.ratio'
+    unit = '%'
+    sample_type = sample.TYPE_DELTA
+
+
+class MysqlBufferPoolSizePollster(DataBaseNotificationBase):
+    """Listen for Trove notifications.
+       Listen in order to mediate with the metering framework.
+    """
+    resource_name = 'mysql.buffer.pool.size'
+    unit = 'MB'
+    sample_type = sample.TYPE_GAUGE
+
+
+class MysqlBufferPoolReadHitsRatioPollster(DataBaseNotificationBase):
+    """Listen for Trove notifications.
+       Listen in order to mediate with the metering framework.
+    """
+    resource_name = 'mysql.buffer.pool.read.hits.ratio'
+    unit = '%'
+    sample_type = sample.TYPE_DELTA
+
+
+class MysqlQcacheQueryHitsRatioPollster(DataBaseNotificationBase):
+    """Listen for Trove notifications.
+       Listen in order to mediate with the metering framework.
+    """
+    resource_name = 'mysql.qcahce.query.hits.ratio'
+    unit = '%'
+    sample_type = sample.TYPE_DELTA
+
+
+class MysqlCachedConnectionsHitsRatioPollster(DataBaseNotificationBase):
+    """Listen for Trove notifications.
+       Listen in order to mediate with the metering framework.
+    """
+    resource_name = 'mysql.cached.connections.hits.ratio'
+    unit = '%'
+    sample_type = sample.TYPE_DELTA
+
+
+class MysqlThreadsRunningPollster(DataBaseNotificationBase):
+    """Listen for Trove notifications.
+       Listen in order to mediate with the metering framework.
+    """
+    resource_name = 'mysql.threads.running'
+    unit = 'connect'
+    sample_type = sample.TYPE_DELTA
+
+
+class MysqlMaxConnectionsPollster(DataBaseNotificationBase):
+    """Listen for Trove notifications.
+       Listen in order to mediate with the metering framework.
+    """
+    resource_name = 'mysql.max.connections'
+    unit = 'connect'
+    sample_type = sample.TYPE_GAUGE
+
+
+class MysqlTransactionCommitPollster(DataBaseNotificationBase):
+    """Listen for Trove notifications.
+       Listen in order to mediate with the metering framework.
+    """
+    resource_name = 'mysql.transaction.commit'
+    unit = 'request'
+    sample_type = sample.TYPE_CUMULATIVE
+
+
+class MysqlTransactionRollbackPollster(DataBaseNotificationBase):
+    """Listen for Trove notifications.
+       Listen in order to mediate with the metering framework.
+    """
+    resource_name = 'mysql.transaction.rollback'
+    unit = 'request'
+    sample_type = sample.TYPE_CUMULATIVE
+
+
+class MysqlTotalSelectPollster(DataBaseNotificationBase):
+    """Listen for Trove notifications.
+       Listen in order to mediate with the metering framework.
+    """
+    resource_name = 'mysql.total.select'
+    unit = 'request'
+    sample_type = sample.TYPE_CUMULATIVE
+
+
+class MysqlReplaceQPSPollster(DataBaseNotificationBase):
+    """Listen for Trove notifications.
+       Listen in order to mediate with the metering framework.
+    """
+    resource_name = 'mysql.replace.qps'
+    unit = 'request'
+    sample_type = sample.TYPE_CUMULATIVE
+
+
+class MysqlReplaceQPSRatePollster(DataBaseNotificationBase):
+    """Listen for Trove notifications.
+       Listen in order to mediate with the metering framework.
+    """
+    resource_name = 'mysql.replace.qps.rate'
+    unit = 'request/s'
+    sample_type = sample.TYPE_DELTA
diff --git a/setup.cfg b/setup.cfg
index 819dd7d..2873bda 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -77,6 +77,27 @@ ceilometer.notification =
 	hardware.ipmi.voltage = ceilometer.ipmi.notifications.ironic:VoltageSensorNotification
 	hardware.ipmi.current = ceilometer.ipmi.notifications.ironic:CurrentSensorNotification
 	hardware.ipmi.fan = ceilometer.ipmi.notifications.ironic:FanSensorNotification
+	mysql.total.connections = ceilometer.database.mysql:MysqlTotalConnectPollster
+	mysql.threads.connections = ceilometer.database.mysql:MysqlThreadsConnectPollster
+	mysql.qps = ceilometer.database.mysql:MysqlQPSPollster
+	mysql.qps.rate = ceilometer.database.mysql:MysqlQPSRatePollster
+	mysql.tps = ceilometer.database.mysql:MysqlTPSPollster
+	mysql.tps.rate = ceilometer.database.mysql:MysqlTPSRatePollster
+	mysql.slow.querys = ceilometer.database.mysql:MysqlSlowQuerysPollster
+	mysql.sync.delay = ceilometer.database.mysql:MysqlSyncSlaveDelayPollster
+	mysql.scan.full.table = ceilometer.database.mysql:MysqlScanFullTablePollster
+	mysql.buffer.pool.dirty.ratio = ceilometer.database.mysql:MysqlBufferPoolDirtyRatioPollster
+	mysql.buffer.pool.size = ceilometer.database.mysql:MysqlBufferPoolSizePollster
+	mysql.buffer.pool.read.hits.ratio = ceilometer.database.mysql:MysqlBufferPoolReadHitsRatioPollster
+	mysql.qcahce.query.hits.ratio = ceilometer.database.mysql:MysqlQcacheQueryHitsRatioPollster
+	mysql.cached.connections.hits.ratio = ceilometer.database.mysql:MysqlCachedConnectionsHitsRatioPollster
+	mysql.threads.running = ceilometer.database.mysql:MysqlThreadsRunningPollster
+	mysql.max.connections = ceilometer.database.mysql:MysqlMaxConnectionsPollster
+	mysql.transaction.commit = ceilometer.database.mysql:MysqlTransactionCommitPollster
+	mysql.transaction.rollback = ceilometer.database.mysql:MysqlTransactionRollbackPollster
+	mysql.total.select = ceilometer.database.mysql:MysqlTotalSelectPollster
+	mysql.replace.qps = ceilometer.database.mysql:MysqlReplaceQPSPollster
+	mysql.replace.qps.rate = ceilometer.database.mysql:MysqlReplaceQPSRatePollster
 ceilometer.discover = 
 	local_instances = ceilometer.compute.discovery:InstanceDiscovery
 	endpoint = ceilometer.central.discovery:EndpointDiscovery
-- 
2.7.4

